lib/rules/id-blacklist.js

```diff
@@ -34,7 +34,7 @@ module.exports = {
         // Helpers
         //--------------------------------------------------------------------------
 
-        var blacklist = context.options;
+        var blacklistRegex = new RegExp("^(" + context.options.join("|") + ")$");
```

Comment:
Umm.. I think we should follow the same pattern as `id-match` and not include start or end of the string in the RegEx. Doesn't take users too long to add it themselves, and there might be some edge cases where they want to check for stuff in the middle of the name. For example blacklisting all of the swearing anywhere in the variable name declaration.